From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Tue, 5 Jul 2022 14:03:12 +1000
Subject: [PATCH] Fix decompile issues

Original Patch Found On Multipaper Which Is Licensed Under The MIT
Remodified By Pancakse

diff --git a/src/main/java/net/minecraft/commands/synchronization/ArgumentTypeInfos.java b/src/main/java/net/minecraft/commands/synchronization/ArgumentTypeInfos.java
index 7dd1c710267db2b299442ca4be33046328d18091..ee9f7bb0e3db19dda3e6300a0a6670e558314cd7 100644
--- a/src/main/java/net/minecraft/commands/synchronization/ArgumentTypeInfos.java
+++ b/src/main/java/net/minecraft/commands/synchronization/ArgumentTypeInfos.java
@@ -127,8 +127,8 @@ public class ArgumentTypeInfos {
         return register(registry, "uuid", UuidArgument.class, SingletonArgumentInfo.contextFree(UuidArgument::uuid));
     }
 
-    private static <T extends ArgumentType<?>> Class<T> fixClassType(Class<? super T> clazz) {
-        return clazz;
+    private static <T extends ArgumentType<?>> Class<T> fixClassType(Class<? extends T> clazz) { // IceCream - fix decompile
+        return (Class<T>) clazz; // IceCream - fix decompile
     }
 
     public static boolean isClassRecognized(Class<?> clazz) {
@@ -140,7 +140,7 @@ public class ArgumentTypeInfos {
         if (argumentTypeInfo == null) {
             throw new IllegalArgumentException(String.format(Locale.ROOT, "Unrecognized argument type %s (%s)", argumentType, argumentType.getClass()));
         } else {
-            return argumentTypeInfo;
+            return (ArgumentTypeInfo<A, ?>) argumentTypeInfo; // IceCream - fix decompile
         }
     }
 
